src/SerializerBuilder.php

```diff
@@ -639,7 +639,7 @@ private function decorateAnnotationReader(Reader $annotationReader): Reader
             if (class_exists(FilesystemAdapter::class)) {
                 $annotationsCache = new FilesystemAdapter('', 0, $this->cacheDir . '/annotations');
                 $annotationReader = new PsrCachedReader($annotationReader, $annotationsCache, $this->debug);
-            } else {
+            } elseif (class_exists(FilesystemCache::class) && class_exists(CachedReader::class)) {
```

Comment:
* `FilesystemCache` is part of Doctrine Cache 1 which is unmaintained now. Doctrine Cache 2 is a transitional package with PSR-6 wrappers only. It does not contain any Cache implementations anymore.

* `CachedReader` from the Doctrine Annotations package has been removed in Annotations 2. Since you don't support Annotations 2 yet, the second check is technically not necessary yet.